core/lib/vm_executor/src/oneshot/contracts.rs

```diff
@@ -22,11 +65,11 @@ pub(super) struct MultiVMBaseSystemContracts {
     vm_1_5_0_small_memory: BaseSystemContracts,
     /// Contracts to be used after the 1.5.0 upgrade
     vm_1_5_0_increased_memory: BaseSystemContracts,
+    _contracts_kind: PhantomData<fn() -> C>,
```

Comment:
dumb question: why not just `PhantomData<C>`?

Reply:
Here specifically, it's defensive programming out of habit 🙃 Overall, using `PhantomData<fn() -> C>`, `PhantomData<fn(C)>` etc. determines a couple of related properties, such as the layout requirements of the generic type (`PhantomData<C>` has the same layout reqs as `C`) and [variance](https://doc.rust-lang.org/nomicon/subtyping.html#variance).